### ISSUE TITLE
Feature request: shape

### ISSUE THREAD

**Issue by bcarry** [NASA-Planetary-Science/sbpy#229]
This is a request for 
* [X] a new feature
* [ ] an enhancement to existing sbpy functionality
* [ ] somethings else: [explain here]

The requested changes will be implemented by
* [X] me
* [X] the sbpy developers 

**High-level concept**
Suite of tools to compute the orientation of a shape model from an observer point of view, giving a shape/spin, and an Ephem object

**Explain the relevance to sbpy**
This is very specific to small bodies. 

**Proposal details**
-Definition of objects for spin and for shape models
 -Spin:
   -Pole coordinates (quantity, associated with reference frame EC/EQ @ epoch)
   -Sidereal rotation period (quantity)
   -There should be method on that object to convert from EC to EQ (LC generally community uses EC, radar uses EQ)
   -There should be method on that object to compute obliquity (often requested), can be done with spin and data.orbit as input
  -Shape:
   -Vertices (x,y,z coordinates in cartesian frame linked with sb)
   -Facet (list of connections between vertices)
-Codes to read spin and shape from files (typical formats from DAMIT for instance : https://astro.troja.mff.cuni.cz/projects/asteroids3D) and store them in the spin/shape objects
-Codes to compute subobserver and subsolar points (require spin and Ephem objects), and pole angle

**Example (pseudo-)code**
No idea yet: as said during the sbpy/EPSC splinter, I am willing to contribute, I have IDL codes that do the computations mentioned above, will convert them to python, but need help to make it sbpy-compliant

**Comment by scraptechguy** [comment]
Hey, is anyone still working on this?

**Comment by bcarry** [comment]
I am not.
Back then my python was too limited to contribute and did not receive feedback.
I have no time availability now for that.

**Comment by mkelley** [comment]
Hi, I had a discussion with Josef Durech at the ACM meeting last month, who is also interested in contributing to shape model work.  This aspect of sbpy is, as far as I know, essentially undefined, but this issue does provide a good starting point.  I think what we need to do is develop the API before any real coding is done.

@jianyangli is our lead for shape models.  Do you have any comments to add?

**Comment by jianyangli** [comment]
Yes this task is definitely on the sbpy list.  We should keep this issue open to track progress and facilitate discussion.  I think we should probably have a zoom meeting with relevant contributors to discuss the functionalities and API of the submodule.
